adal/src/main/java/com/microsoft/aad/adal/Discovery.java

```diff
@@ -104,16 +104,17 @@ public boolean isValidAuthority(URL authorizationEndpoint) {
                         ADALError.DEVELOPER_AUTHORITY_CAN_NOT_BE_VALIDED,
                         new AuthenticationException(ADALError.DISCOVERY_NOT_SUPPORTED));
                 return false;
-            } else if (sValidHosts.contains(authorizationEndpoint.getHost().toLowerCase(Locale.US))) {
```

Comment:
just trying to reduce the number of early returns. do you think this is unreadable? @weijjia 

---
In reply to: [66004394](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/652#discussion_r66004394) [](ancestors = 66004394)